digitalocean/resource_digitalocean_monitor_alert.go

```diff
@@ -0,0 +1,323 @@
+
+	for _, slackChannel := range slackChannels {
+		slack := slackChannel.(map[string]interface{})
+		n := godo.SlackDetails{
+			Channel: slack["channel"].(string),
+			URL:     slack["url"].(string),
+		}
+
+		expandedSlackChannels = append(expandedSlackChannels, n)
+	}
+
+	return expandedSlackChannels
+}
+
+func flattenSlack(slackChannels []godo.SlackDetails) []interface{} {
+	result := make([]map[string]interface{}, 0, len(slackChannels))
+
+	for _, slackChannel := range slackChannels {
+		item := make(map[string]interface{})
+		item["url"] = slackChannel.URL
+		item["channel"] = slackChannel.Channel
+		result = append(result, item)
+	}
+
+	return []interface{}{result}
+}
+
+func expandEmail(config []interface{}) []string {
+	if len(config) == 0 {
+		return nil
+	}
+	emailList := make([]string, len(config))
+
+	for i, v := range config {
+		emailList[i] = v.(string)
+	}
+
+	return emailList
+}
+
+func flattenEmail(emails []string) *schema.Set {
+	if len(emails) == 0 {
+		return nil
+	}
+
+	flattenedEmails := schema.NewSet(HashStringIgnoreCase, []interface{}{})
+	for _, v := range emails {
+		if v != "" {
+			flattenedEmails.Add(v)
+		}
+	}
+
+	return flattenedEmails
+}
+
+func expandEntities(config []interface{}) []string {
+	alertEntities := make([]string, len(config))
+
+	for i, v := range config {
+		alertEntities[i] = v.(string)
+	}
+
+	return alertEntities
+}
+
+func flattenEntities(entities []string) []interface{} {
+	// it seems there are many functions like this in different places in the code base.
+	// maybe a utility library would be better
+	if len(entities) == 0 {
+		return nil
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"entities": entities,
+		},
+	}
+}
+
+func resourceDigitalOceanMonitorAlertUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*CombinedConfig).godoClient()
+	opts := &godo.AlertPolicyUpdateRequest{}
+
+	if d.HasChange("alerts") {
+		alerts := expandAlerts(d.Get("alerts").([]interface{}))
+		opts.Alerts = alerts
+		client.Monitoring.UpdateAlertPolicy(ctx, d.Id(), opts)
+	}
+	// what more to update here?
+
+	return resourceDigitalOceanMonitorAlertRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanMonitorAlertRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*CombinedConfig).godoClient()
+
+	alert, resp, err := client.Monitoring.GetAlertPolicy(ctx, d.Id())
+
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			log.Printf("[DEBUG] Alert (%s) was not found - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading Alert: %s", err)
+	}
+
+	d.SetId(alert.UUID)
+	d.Set("description", alert.Description)
+	d.Set("enabled", alert.Enabled)
+	d.Set("compare", alert.Compare)
+	d.Set("alerts", flattenAlerts(alert.Alerts))
+	d.Set("value", alert.Value)
+	d.Set("window", alert.Window)
+	d.Set("entities", flattenEntities(alert.Entities))
```

Comment:
This can be simplified since `alert.Entities` is already `[]string`.

```suggestion
	d.Set("entities", alert.Entities)
```